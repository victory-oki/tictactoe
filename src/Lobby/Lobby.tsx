import React, { useEffect } from 'react'
import { NavigateFunction, useNavigate, useParams } from 'react-router-dom'
import { ContainerStyled } from './LobbyStyles'
import { ProgressSpinner } from 'primereact/progressspinner';
import { useSocket } from '../socketContext';
import { EVENTS } from '../gameModel';
function Lobby() {
    const { id: roomId } = useParams();
    const socket = useSocket();
    const navigate: NavigateFunction = useNavigate();

    useEffect(() => {
        setupSocketListeners();
        return ()=>{
          destroyListeners();
        }
      }, [roomId]);

    const setupSocketListeners = ()=>{
        socket.on(EVENTS.GAME_START, ()=>{
            navigate(`/game/${roomId}`, { replace: false });
        })
    }

    const destroyListeners = ()=>{
        socket.off(EVENTS.GAME_START)
    }

    return (
        <ContainerStyled>
            <div className="container">
                <div className="header">
                    <h1 className="header__title">Lobby</h1>
                    <p className="header__brief">We are waiting for your friend, please share the following code: <b className='u-emphasis'><code data-testid = 'code-section'>{roomId?.toLocaleUpperCase()}</code></b></p>
                    <p className="brief">you would be redirected as soon as your friend joins</p>
                </div>
                <div className="content">
                <ProgressSpinner style={{width: '150px', height: '150px'}} strokeWidth="6" fill="transparent" animationDuration=".5s"/>
                </div>
            </div>
        </ContainerStyled>
    )
}

export default Lobby
