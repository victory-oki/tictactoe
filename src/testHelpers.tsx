
import React from "react";
import TestRenderer from 'react-test-renderer';
import * as ShallowRenderer from 'react-test-renderer/shallow';

import { BrowserRouter as Router } from 'react-router-dom';
import { store } from "./store";
import { Provider } from "react-redux";

export const testSnapshot = (component: JSX.Element, shallow = false) => {
    const tree = shallow ?
      ShallowRenderer.createRenderer().render(component) :
      TestRenderer.create(component).toJSON()
    ;
    expect(tree).toMatchSnapshot();
};


export const wrapContext = (ui: JSX.Element) => (
    <Router>
        <Provider store={store}>
            {ui}
        </Provider>
    </Router>
)
