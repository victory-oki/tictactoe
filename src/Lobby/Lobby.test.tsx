
import { render, screen, within } from "@testing-library/react";
import { testSnapshot, wrapContext } from "../testHelpers"
import Lobby from "./Lobby";
import * as routerMock from 'react-router-dom';

jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useParams: ()=>({id:'TESTCODE'}),
}));

describe('Test if lobby renders correctly',()=>{

    test('Renders Correctly with params', ()=>{

        const wrappedComponent: JSX.Element = wrapContext(<Lobby />);
        render(
            wrappedComponent
        );
        within(screen.getByTestId('code-section')).getByText('TESTCODE');
    })

    it('renders correctly', ()=>{
        const wrappedComponent: JSX.Element = wrapContext(<Lobby />);
        testSnapshot(wrappedComponent)
    })
})
