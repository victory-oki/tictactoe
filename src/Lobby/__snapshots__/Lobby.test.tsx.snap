// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if lobby renders correctly renders correctly 1`] = `
<div
  className="sc-bczRLJ zPssV"
>
  <div
    className="container"
  >
    <div
      className="header"
    >
      <h1
        className="header__title"
      >
        Lobby
      </h1>
      <p
        className="header__brief"
      >
        We are waiting for your friend, please share the following code: 
        <b
          className="u-emphasis"
        >
          <code
            data-testid="code-section"
          >
            TESTCODE
          </code>
        </b>
      </p>
      <p
        className="brief"
      >
        you would be redirected as soon as your friend joins
      </p>
    </div>
    <div
      className="content"
    >
      <div
        aria-busy={true}
        className="p-progress-spinner"
        id={null}
        role="alert"
        style={
          Object {
            "height": "150px",
            "width": "150px",
          }
        }
      >
        <svg
          className="p-progress-spinner-svg"
          style={
            Object {
              "animationDuration": ".5s",
            }
          }
          viewBox="25 25 50 50"
        >
          <circle
            className="p-progress-spinner-circle"
            cx="50"
            cy="50"
            fill="transparent"
            r="20"
            strokeMiterlimit="10"
            strokeWidth="6"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
